File: ci/mkl_test.c
#include <stdio.h>
#include <mkl.h>

int main() {
    double a[6] = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0};
    double b[6] = {1.0, 1.0, 1.0, 1.0, 1.0, 1.0};
    double c[4];
    
    // C = A * B where A is 2x3, B is 3x2, C is 2x2
    cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans,
               2, 2, 3, 1.0, a, 3, b, 2, 0.0, c, 2);
    
    printf("MKL BLAS test call successful\n");
    
    return 0;
}

File: ci/mkl_test.f90
program mkl_test
    implicit none
    double precision :: a(2,3), b(3,2), c(2,2)
    integer :: i, j
    
    ! Initialize matrices
    do i = 1, 2
        do j = 1, 3
            a(i,j) = (i-1)*3 + j
        end do
    end do
    
    b = 1.0d0
    c = 0.0d0
    
    ! C = A * B
    call dgemm('N', 'N', 2, 2, 3, 1.0d0, a, 2, b, 3, 0.0d0, c, 2)
    
    print *, 'MKL BLAS Fortran test successful'
    print *, 'Result:', c
end program mkl_test
